src/client/utils/url.js

```diff
@@ -13,11 +19,17 @@ export function getProxyUrl (url, proxyHostname, proxyPort, sessionId, resourceT
     // NOTE: Resolves relative URLs.
     url = destLocation.resolveUrl(url);
 
+    proxyHostname = proxyHostname || location.hostname;
+    proxyPort     = proxyPort || location.port.toString();
+    sessionId     = sessionId || settings.get().sessionId;
+
     // NOTE: If the relative URL contains no slash (e.g. 'img123'), the resolver will keep
     // the original proxy information, so that we can return such URL as is.
     // TODO: Implement the isProxyURL function.
     var parsedAsProxy   = sharedUrlUtils.parseProxyUrl(url);
-    var isValidProxyUrl = !!parsedAsProxy;
+    var isValidProxyUrl = !!parsedAsProxy && parsedAsProxy.proxy.hostname === proxyHostname &&
```

Comment:
It looks strange for me that `parsedAsProxy` has some fields because. I expect a boolean value from this variable.
Maybe we should rename it to `parsedProxyUrl` or something else? /cc @LavrovArtem @churkin